eth/catalyst/api.go

```diff
@@ -488,8 +488,11 @@ func (api *ConsensusAPI) forkchoiceUpdated(update engine.ForkchoiceStateV1, payl
 
 			// Write L1Origin.
 			rawdb.WriteL1Origin(api.eth.ChainDb(), l1Origin.BlockID, l1Origin)
-			// Write the head L1Origin.
-			rawdb.WriteHeadL1Origin(api.eth.ChainDb(), l1Origin.BlockID)
+
+			// Write the head L1Origin, only when it's not a preconfirmation block.
+			if !l1Origin.IsPreconfBlock() {
```

Comment:
Why not check preconf hardfork by L2 block number?

Reply:
Don't think its necessary?

Reply:
The [setting](https://github.com/taikoxyz/taiko-mono/pull/18746#discussion_r1935682835) here may be wrong.

The check that is not dependent on parameters may be safer.